v3/src/lib/use-cloud-file-manager.ts

```diff
@@ -9,6 +9,128 @@ import { appState } from "../models/app-state"
 import { createCodapDocument, isCodapDocument } from "../models/codap/create-codap-document"
 import { t } from "../utilities/translation/translate"
 import { removeDevUrlParams } from "../utilities/url-params"
+import { Logger } from "./logger"
+
+const locales = [
+  {
+    langName: 'Deutsch',
+    langDigraph: 'de',
+    countryDigraph: 'DE',
+    icon: 'flag flag-de'
+  },
+  {
+    langName: 'English',
+    langDigraph: 'en',
+    countryDigraph: 'US',
+    icon: 'flag flag-us'
+  },
+  {
+    langName: 'Español',
+    langDigraph: 'es',
+    countryDigraph: 'ES',
+    icon: 'flag flag-es'
+  },
+  {
+    langName: 'فارسی',
+    langDigraph: 'fa',
+    countryDigraph: 'IR',
+    icon: 'flag flag-ir'
+  },
+  {
+    langName: 'Ελληνικά',
+    langDigraph: 'el',
+    countryDigraph: 'GR',
+    icon: 'flag flag-gr'
+  },
+  {
+    langName: 'עברית',
+    langDigraph: 'he',
+    countryDigraph: 'IL',
+    icon: 'flag flag-il'
+  },
+  {
+    langName: '日本語',
+    langDigraph: 'ja',
+    countryDigraph: 'JP',
+    icon: 'flag flag-jp'
+  },
+  {
+    langName: '한국어',
+    langDigraph: 'ko',
+    countryDigraph: 'KO',
+    icon: 'flag flag-kr'
+  },
+  {
+    langName: 'Bokmål',
+    langDigraph: 'nb',
+    countryDigraph: 'NO',
+    icon: 'flag flag-no'
+  },
+  {
+    langName: 'Nynorsk',
+    langDigraph: 'nn',
+    countryDigraph: 'NO',
+    icon: 'flag flag-no'
+  },
+  {
+    langName: 'Polski',
+    langDigraph: 'pl',
+    countryDigraph: 'PL',
+    icon: 'flag flag-pl'
+  },
+  {
+    langName: 'Português do Brasil',
+    langDigraph: 'pt-BR',
+    countryDigraph: 'BR',
+    icon: 'flag flag-br'
+  },
+  {
+    langName: 'ไทย',
+    langDigraph: 'th',
+    countryDigraph: 'TH',
+    icon: 'flag flag-th'
+  },
+  {
+    langName: 'Türkçe',
+    langDigraph: 'tr',
+    countryDigraph: 'TR',
+    icon: 'flag flag-tr'
+  },
+  {
+    langName: '繁体中文',
+    langDigraph: 'zh-TW',
+    countryDigraph: 'TW',
+    icon: 'flag flag-tw'
+  },
+  {
+    langName: '简体中文',
+    langDigraph: 'zh-Hans',
+    countryDigraph: 'Hans',
+    icon: 'flag flag-cn'
+  }
+]
+function makeNewLocaleUrl(digraph: string, locationHref: string) {
+  const location = new URL(locationHref)
+  const locMatch = location.pathname.match(/(^.*\/static\/dg\/)[^/]+(\/cert\/.*$)/)
+  let baseURL = "https://codap3.concord.org/branch/main/"
```

Comment:
I assume this `baseURL` basically never comes up, but do we want it to be a branch link? Should it just be `https://codap3.concord.org`?